lib/health_monitor/monitor.rb

```diff
@@ -31,15 +34,19 @@ def check(request: nil)
   def provider_result(provider, request)
     monitor = provider.new(request: request)
     monitor.check!
-
     {
       name: provider.provider_name,
       message: '',
       status: STATUSES[:ok]
     }
+  rescue HealthWarning => e
+    {
+      name: provider.provider_name,
+      message: e.message,
+      status: STATUSES[:warn]
+    }
```

Comment:
In addition, make sure that a warning path responds with an appropriate status - not error.